### ISSUE TITLE
`used-before-assignment` false positive for named expressions in a not-first position in container

### ISSUE THREAD

**Issue by jacobtylerwalls** [PyCQA/pylint#8252]
This raises `used-before-assignment`:
```python
def expression_in_ternary_operator_inside_container_later_position():
    """Named expression follows unrelated item in container"""
    return [23, val3 if (val3 := 'something') else 'anything']
```

***

The check for `BaseContainer` should also check all elements. I can handle that in another PR.

_Originally posted by @jacobtylerwalls in https://github.com/PyCQA/pylint/pull/8176#pullrequestreview-1292221144_
